lib/libzfs/libzfs_import.c

```diff
@@ -77,7 +77,7 @@ refresh_config(libzfs_handle_t *hdl, nvlist_t *config)
 	if (zcmd_write_conf_nvlist(hdl, &zc, config) != 0)
 		return (NULL);
 
-	dstbuf_size = MAX(CONFIG_BUF_MINSIZE, zc.zc_nvlist_conf_size * 4);
+	dstbuf_size = MAX(CONFIG_BUF_MINSIZE, zc.zc_nvlist_conf_size * 32);
```

Comment:
I believe the original concern was to avoid asking the kernel to perform an unnecessarily large allocation.  If the `vmem_alloc()` in `get_nvlist()` were to fail then import would fail.  Since `vmalloc()` is used under the hood for this on Linux that's probably not a major concern, _except_ for 32-bit x86 systems which usually only have 128MB of virtual kernel address space for the entire system.  In practice, how large does this value end up being with the new multiplier for a large pool?

Reply:
Does the kernel really need to allocate for that?  I thought it was just for the userland buffer.  At any rate, on my test system with 18 top-level vdevs, the actual size is around 750 kB.

Reply:
The userland pages are passed in and then a [copy is made in kernel space](https://github.com/openzfs/zfs/blob/master/module/zfs/zfs_ioctl.c#L1274-L1307) and that's used to unpack the nvlist.  I presume it was originally done this way to keep things simple.  We could map the user pages but it's probably not worthwhile.  750 kB still seems reasonable to me since it's so short lived.